### PR TITLE
Transport integration

### PR DESCRIPTION
**Pull request by Lezek123**
This PR integrates `ProposalDetials` and related components (like `Details`, `Body`, `VotingSection`, `Votes` etc.) with the transport layer & extrinsics.

It also introduces some minor changes to `ProposalPreviewList` / `ProposalPreview` (since `ProposalPreview` also uses `Details` etc.)

### CODE REVIEW

File: packages/joy-proposals/src/Proposal/Body.tsx [hunk 1]
```diff
@@ -1,16 +1,103 @@
...
+  );
+}
+
+// The methods for parsing params by Proposal type.
+// They take the params as array and return { LABEL: VALUE } object.
+const paramParsers: { [x in ProposalType]: (params: any[]) => { [key: string]: string | number | JSX.Element } } = {
+  "Text" : ([content]) => ({
+    "Content": content,
+  }),
+  "RuntimeUpgrade" : ([wasm]) => {
```
Comment (Gamaranto) [inline packages/joy-proposals/src/Proposal/Body.tsx:32]: Quotes on the keys here are not necessary

File: packages/joy-proposals/src/Proposal/Details.tsx [hunk 2]
```diff
@@ -1,42 +1,60 @@
...
+            <Item.Extra>Substage:</Item.Extra>
+            <Header as="h4">{ substage }</Header>
+          </Item.Content>
+        </Item>
+      ) }
+      { (expiresIn !== null) && (
+        <Item>
+          <Item.Content>
+            <Item.Extra>Expires in:</Item.Extra>
+            <Header as="h4">{`${ expiresIn } blocks`}</Header>
```
Comment (Gamaranto) [inline packages/joy-proposals/src/Proposal/Details.tsx:54]: You can use `expiresIn.toLocaleString("en-US")` to display the block numbers as they were in the wireframe.
Ex: `89478` becomes `89,478`

File: packages/joy-proposals/src/Proposal/Details.tsx [hunk 3]
```diff
@@ -1,42 +1,60 @@
...
   return (
     <Item.Group className="details-container">
       <Item>
         <Item.Content>
           <Item.Extra>Proposed By:</Item.Extra>
-          <Image src={proposal.proposer.avatar_uri} avatar floated="left" />
-          <Header as="h4">{proposal.proposer.about}</Header>
-          <Item.Extra>{proposal.createdAt.toUTCString()}</Item.Extra>
+          { proposer.avatar_uri ?
+            <Image src={ proposer.avatar_uri } avatar floated="left" />
```
Comment (Gamaranto) [inline packages/joy-proposals/src/Proposal/Details.tsx:24]: Currently the handle is not vertically aligned with the Identicon/Image, we can do so by wrapping them both in a flex container and adding some margin to the handle.
```javascript
          <div style={{ display: "flex", alignItems: "center" }}>
            {proposer.avatar_uri ? ... } // Same as before
            <Header as="h4" style={{ margin: "auto 0 auto 0.5rem" }}>
              {proposer.handle}
            </Header>
          </div>
```

File: packages/joy-proposals/src/Proposal/VotingSection.tsx [hunk 4]
```diff
@@ -1,50 +1,101 @@
...
+    () => transport.voteByProposalAndMember(proposalId, memberId),
+    undefined
+  );
+
+  if (vote === undefined) {
+    // Loading / error
+    return null;
+  }
+
+  const voteStr: VoteKindStr | null = voted ? voted : (vote && vote.type.toString() as VoteKindStr);
```
Comment (Gamaranto) [inline packages/joy-proposals/src/Proposal/VotingSection.tsx:69]: Typescript complains here about the ternary assignment being redundant. You can shorten this and by using a short-circuit evaluation:
```javascript
voted || (vote && vote.type.toString() as VoteKindStr);
```

File: packages/joy-proposals/src/runtime/transport.substrate.ts [hunk 5]
```diff
@@ -73,7 +73,7 @@ export class SubstrateTransport extends Transport {
   async proposalById(id: ProposalId): Promise<ParsedProposal> {
     const rawDetails = (await this.proposalDetailsById(id)).toJSON() as { [k: string]: any };
     const type = Object.keys(rawDetails)[0] as ProposalType;
-    const details = rawDetails[type];
```
Comment (Gamaranto) [inline packages/joy-proposals/src/runtime/transport.substrate.ts:76]: This caused a bug with the text proposal. Great catch :)

File: packages/joy-utils/src/TxButton.tsx [hunk 6]
```diff
@@ -130,7 +130,7 @@ function MockTxButton (props: BasicButtonProps) {
 function ResolvedButton (props: BasicButtonProps) {
   const isMock = useTransportContext() instanceof MockTransport;
 
-  return isMock 
+  return isMock
```
Comment (Gamaranto) [inline packages/joy-utils/src/TxButton.tsx:133]: I think this is just an automatic formatting change

File: packages/joy-proposals/src/Proposal/Body.tsx (hunk 1 above)
Comment (Lezek123) [reply packages/joy-proposals/src/Proposal/Body.tsx]: Updated

File: packages/joy-proposals/src/Proposal/Details.tsx (hunk 2 above)
Comment (Lezek123) [reply packages/joy-proposals/src/Proposal/Details.tsx]: Updated

File: packages/joy-proposals/src/Proposal/Details.tsx (hunk 3 above)
Comment (Lezek123) [reply packages/joy-proposals/src/Proposal/Details.tsx:24]: As you mentioned - we can move the UI fixes to another PR

File: packages/joy-proposals/src/Proposal/VotingSection.tsx (hunk 4 above)
Comment (Lezek123) [reply packages/joy-proposals/src/Proposal/VotingSection.tsx]: Great, my editor didn't catch it, fixed!

**Pull request merged**